core/trino-main/src/main/java/io/trino/util/JsonUtil.java

```diff
@@ -105,6 +106,9 @@
 
 public final class JsonUtil
 {
+    // StringReader outperforms InputStreamReader for small inputs. Limit based on Jackson benchmarks {@link https://github.com/FasterXML/jackson-benchmarks/pull/9}
+    private static final int STRING_READER_LENGTH_LIMIT = 8192;
```

Comment:
How about leaving a code comment to prevent the same mistake in the future?

Reply:
Makes sense.

Reply:
I'll push once the CI is done.